webapp/channels/src/components/new_search/search_box_hints.tsx

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {useSelector} from 'react-redux';
+
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+
+import type {ProviderResult} from 'components/suggestion/provider';
+import SearchDateSuggestion from 'components/suggestion/search_date_suggestion';
+
+import ErrorBoundary from 'plugins/pluggable/error_boundary';
+
+import type {GlobalState} from 'types/store';
+
+import SearchHints from './search_hint';
+
+type Props = {
+    searchTerms: string;
+    setSearchTerms: (searchTerms: string) => void;
+    searchType: string;
+    selectedOption: number;
+    providerResults: ProviderResult<unknown>|null;
+    focus: (pos: number) => void;
+}
+
+const SearchBoxHints = ({searchTerms, setSearchTerms, searchType, providerResults, selectedOption, focus}: Props) => {
+    const license = useSelector(getLicense);
+    const filterSelectedCallback = useCallback((filter: string) => {
+        if (searchTerms.endsWith(' ') || searchTerms.length === 0) {
+            setSearchTerms(searchTerms + filter);
+            focus(searchTerms.length + filter.length);
```

Comment:
Could focusing the textbox and moving the cursor always be part of calling `setSearchTerms`? It seems like `setSearchTerms` is always followed by `focus` in this component

Reply:
It has changed, and it is not always like that anymore.